### ISSUE TITLE
Other issues

### ISSUE THREAD

**Issue by GetContented** [rwobben/paytime#7]
Ok, so you wrote a list of the 3 things you'd like to have... here is a set of ideas and the order that I'd planned out a few days ago... thinking as I did of the order that things could be taught in...

-- adjust the types so that they are data types
-- adjust an Account so it actually holds a list of transactions data Account = Account [Transaction]
-- use records, and make sure that accounts have a name (introduce String)
-- write a function to retrieve the last Transaction on an Account.
-- write a function retrieve the last n Transactions on an Account
-- deal with precise amounts rather than Integer or Float
-- Operators vs Postfix functions, precedence
-- work as many list functions in as possible
-- write a function to repeat the last withdrawl again
-- Sum Types, Product Types, what Typeclasses are
-- Monoid, and Semigroup - use on List, String (String / Char)
-- use Maybe to attempt a series of transactions to ensure the account doesn't go below 0, and abort if none worked. (case and pattern matching, then Functor and fmap)
   -- from Control.Monad, use : foldM (\x y -> if (x + y) > 0 then Just (x + y) else Nothing) 0 [1,2,3,4,-15,1,3,5,3,3,300]

-- start Using IO to use a file as a crude database, and then accept input like an ATM, in a loop

-- add Dates to transactions (will need to use the Data.Time / Data.Date and its monad, I think, to determine current time, perhaps).
-- write a function to retrieve the beginning date of an Account

-- add the capacity to have multiple accounts per person, and to have many people
-- add transfers between accounts, and between people
-- add transaction types (ATM / VISA / EFT), and account features [ATM, VISA, DEBT]
-- add transaction fees

**Comment by rwobben** [comment]
I think there are a lot of things I also thought of in the future. 
We can also use that as a sort of path.

**Comment by rwobben** [comment]
Schould I make a isssue of the first three things so we can work on it

**Comment by GetContented** [comment]
Sure, if you like... maybe you can begin to see how to pull these bigger tasks into smaller tasks so they're easier to do. :)

When you can do recursion, folds and use filter, map and other higher order functions, and now that we're starting to put algebraic data types in, we'll naturally start to move towards some of the intermediate things in Haskell. We just have to keep perspective.

**Comment by rwobben** [comment]
oke

**Comment by GetContented** [comment]
Ok I feel like you've got a pretty good handle on the common higher order functions by now.

**Comment by rwobben** [comment]
:) gives me a good feeling to hear that from you. 
Also I came this far by your patience and your way of explanations.
So a compliment for you.

**Comment by GetContented** [comment]
Thanks. I hope it hasn't been too painful. I believe it's possible for people to learn Haskell without pain.
It's been somewhat useful for me.

There's still a lot to learn, but the beginner basics are in place, which is good, and most importantly, you  are confident enough that you can actually write some Haskell code on your own without too much trouble, which is great :)

**Comment by rwobben** [comment]
no, I was at one point that I want to quit but then you were getting me on the right track.
